### DISCUSSION TITLE
Fetch url and sha from json

### DISCUSSION THREAD

**Discussion by pdgendt** [Homebrew/discussions#5301] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9edabf02ce29e54d0b717baa757cff38ad0c0e25
Last commit: 4 days ago
Core tap HEAD: 0a41f841af0b3bacb0491b574c87c6684b521659
Core tap last commit: 3 hours ago
Core tap JSON: 19 Apr 06:24 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 20
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 20-core 64-bit unknown_0x6_0x97
Clang: 14.0.0
Git: 2.37.3 => /bin/git
Curl: 7.81.0 => /bin/curl
Kernel: Linux 5.15.146.1-microsoft-standard-WSL2 x86_64 GNU/Linux
OS: Ubuntu 22.04.4 LTS (jammy)
WSL: 2 (Microsoft Store)
Host glibc: 2.35
/usr/bin/gcc: 11.4.0
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

_Disclaimer: ruby newbie_

Is it possible to have a `head` block fetch a `json` object from a site, and grab the `url` and `sha256` from there?

**Comment by SMillerDev** [comment]
> Is it possible to have a head block fetch a json object from a site, and grab the url and sha256 from there?

No, since `head` by definition points to a CVS system.

**Reply by pdgendt** [reply]
Thanks for the fast response.

Would it be a welcome addition to be able to have `head` follow nightly (successful) builds that way?

**Reply by SMillerDev** [reply, marked as answer]
No, `head` is the `HEAD` of the upstream repository. This sounds like it should be something else, like the `devel` that was removed since it broke all the time.

**Reply by pdgendt** [reply]
Sadly, that does sound like it was something useful indeed 😅

**Reply by pdgendt** [reply]
@SMillerDev would extending the `SoftwareSpec` class with some way of fetching that data be an option? That way both `stable` and `head` can use this "backend".

**Reply by SMillerDev** [reply]
I don't really see a benefit in this, but you're always welcome to make a pull request. Maybe there it'll become clear what the value is

**Reply by pdgendt** [reply]
Maybe some context, but probably it's too much hassle to even start after reading https://github.com/Homebrew/homebrew-core/issues/134298.

I would have liked to install the last **buildable** version of [zig](https://github.com/ziglang/zig), which can be found using the JSON api https://ziglang.org/download/index.json. The map contains a `master` release tarball.
